src/combinatorics/simple_graph/basic.lean

```diff
@@ -576,4 +504,49 @@ end
 
 end complement
 
+section maps
+
+/-- A graph homomorphism is a map on vertex sets that respects adjacency relations -/
+abbreviation homomorphism (G G' : simple_graph V) := rel_hom (G.adj) (G'.adj)
+
+infix ` →g ` : 50 := homomorphism
+
+/-- A graph isomorphism is an bijective map on vertex sets that respects adjacency relations -/
+abbreviation isomorphism (G G' : simple_graph V) := rel_iso G.adj G'.adj
+
+infix ` ≃g ` : 50 := isomorphism
+
+/-- A graph embedding is an embedding `f` such that for vertices `v w : V`,
+  `G.adj f(v) f(w) ↔ G.adj v w `. Its image is an induced subgraph of G' -/
+abbreviation embedding := rel_embedding G.adj G'.adj
+
+lemma graph_iso_self : G ≃g G :=
+begin
+  fconstructor,
+  exact equiv.cast rfl,
+  intros a b,
+  exact iff.rfl,
+end
+
+lemma graph_hom_self : G →g G :=
+begin
+  fconstructor,
+  exact equiv.cast rfl,
+  intros a b,
+  exact (mem_edge_set G).mp,
+end
```

Comment:
These should be `def`, since you are constructing data. A function or an equiv are data because the provide an explicit function that does the mapping, not just a proof that it is bijective etc.

Additionally, the should probably be called something like `graph.isomorphism.refl`.

Reply:
OK I have fixed that.